pkg/platform/kube/project/external/leader/iguazio/client.go

```diff
@@ -0,0 +1,38 @@
+package iguazio
+
+import (
+	"github.com/nuclio/nuclio/pkg/platform"
+	"github.com/nuclio/nuclio/pkg/platform/kube/project/external/leader"
+	"github.com/nuclio/nuclio/pkg/platformconfig"
+
+	"github.com/nuclio/logger"
+	"github.com/nuclio/nuclio-sdk-go"
+)
+
+type Client struct {
+	leader.Client
+
+	logger                logger.Logger
+	platformConfiguration *platformconfig.Config
+}
+
+func NewClient(parentLogger logger.Logger, platformConfiguration *platformconfig.Config) (*Client, error) {
+	newClient := Client{
+		logger:                parentLogger.GetChild("iguazio-leader-client"),
```

Comment:
Since loggers structure is inheritance, you want to leave the "delta" change at the end of its name, so that, when reading logs, you will find the logger "who" field more easier

e.g.:
logger x -> logger y -> logger z prints:
```
{
...
   "who": "x.y.z",
...
}

now, when the name become too long, it is harder to understand who is the "who" and that is why, I suggest to leave the most significant change to the right, so instead of `iguazio-leader-client`, I'd use `client-leader-iguazio`